rest_framework_dataclasses/typing_utils.py

```diff
@@ -131,7 +137,7 @@ def is_optional_type(tp: type) -> bool:
     none_type = type(None)
     return (
         origin is not None and
-        origin is typing.Union and
+        (origin is typing.Union or (UnionType is not None and origin is UnionType)) and
```

Comment:
This is a bit ugly.

I guess I could remove the `UnionType is not None` check, since we just established that `origin is not None`.